### PR TITLE
Allow execution of multiple instances of the same plugin.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

**Comment by PiotrSikora** [comment]
Note: This relands the previous commit and adds `PluginHandle` to allow shutdown-on-destroy semantics.

File: src/wasm.cc [hunk 1]
```diff
@@ -514,24 +499,71 @@ std::shared_ptr<WasmHandleBase> getThreadLocalWasm(std::string_view vm_key) {
...
+    }
+    // Remove stale entry.
+    local_plugins.erase(key);
+  }
+  // Create and initialize new thread-local Plugin.
+  auto wasm_handle = getOrCreateThreadLocalWasm(base_handle, clone_factory);
+  if (!wasm_handle) {
+    return nullptr;
+  }
+  auto plugin_context = wasm_handle->wasm()->start(plugin);
```
Comment (mathetake) [inline src/wasm.cc:550]: nit 
```suggestion
  // Create and initialize new thread-local Plugin.
  auto plugin_context = wasm_handle->wasm()->start(plugin);
```

File: src/wasm.cc [hunk 2]
```diff
@@ -514,24 +499,71 @@ std::shared_ptr<WasmHandleBase> getThreadLocalWasm(std::string_view vm_key) {
...
+  auto it = local_plugins.find(key);
+  if (it != local_plugins.end()) {
+    auto plugin_handle = it->second.lock();
+    if (plugin_handle) {
+      return plugin_handle;
+    }
+    // Remove stale entry.
+    local_plugins.erase(key);
+  }
+  // Create and initialize new thread-local Plugin.
```
Comment (mathetake) [inline src/wasm.cc:545]: nit:
```suggestion
  // Get thread-local WasmVM.
```

**Review by mathetake — APPROVED** [review]
LGTM with nits on comments! Thanks, this now looks much better than the previous iteration

**Pull request merged**